src/services/catalog.service.ts

```diff
@@ -21,23 +21,30 @@ export class CatalogService {
     this.logger = Logger;
   }
 
-  public getCatalogItems() {
-    return this.$q.all({
+  public getCatalogItems(includeTemplates: boolean) {
+    let promises: any = {
       serviceClasses: this.dataService.list({
         group: 'servicecatalog.k8s.io',
         resource: 'serviceclasses'
       }, {}),
-      imageStreams: this.dataService.list("imagestreams", {namespace: "openshift"})
-    }).then((values) => {
+      imageStreams: this.dataService.list("imagestreams", { namespace: "openshift" })
+    };
+
+    if (includeTemplates) {
+      promises.templates = this.dataService.list("templates", { namespace: "openshift" });
+    }
+
+    return this.$q.all(promises).then((values) => {
       let serviceClasses: any = values.serviceClasses.by("metadata.name");
       let imageStreams: any = values.imageStreams.by("metadata.name");
-      return this.convertToServiceItems(serviceClasses, imageStreams);
+      let templates: any = values.templates ? values.templates.by("metadata.name") : {};
+      return this.convertToServiceItems(serviceClasses, imageStreams, templates);
     }, () => {
       this.logger.log("Error Loading Catalog Items");
```

Comment:
I feel like we should not be bombing out if one thing fails to load but the other two succeed, seems like we could at least show what was able to load and then show a message that not everything loaded successfully.  But we can open a separate issue to address this. Note that $q.all fails out at the first failure, so you'd need a different way of handling this.

Just considering the fact that the svc cat api server is separate and could potentially be down even though the master is available.